js/project-map.js

```diff
@@ -90,59 +100,73 @@ const loadMapLayers = () => {
     }, 'place-city-sm')
-    
+
     document.getElementById('Project-Area').innerHTML = formatedData(Math.round(totalArea))
     document.getElementById('Total-Map-Edits').innerHTML = formatedData(Math.round(totalEdits))
     document.getElementById('Community-Mappers').innerHTML = formatedData(Math.round(totalMappers))
     document.getElementById('Countries-Covered').innerHTML = countryList.length
+
     map.on('mousemove', function (e) {
-      var features = map.queryRenderedFeatures(
-        [e.point.x, e.point.y],
-        {layers: ['tm-projects-black-circle', 'tm-projects-symbol']}
-      )
-      if (features.length) {
-        map.getCanvas().style.cursor = 'pointer'
-        $('#hover-details').empty()
-        $('#hover-details').removeClass('hide')
-        $('#hover-details').append(
-          '<p class="hover-name">' +
-          '<a target="_blank" href="https://tasks.hotosm.org/project/' +
-          features[0].properties.id +
-          '">#' + features[0].properties.id + '</a>' +
-            ' - ' +
-            features[0].properties.name + '</p>' +
-            '<p id="proj-details">Click on the project to see more details</p>'
-
-        )
-      } else {
-        map.getCanvas().style.cursor = ''
+      var features = map.queryRenderedFeatures(e.point, {
+        layers: ['tm-projects-black-circle', 'tm-projects-symbol']
+      });
+
+      map.getCanvas().style.cursor = (features.length) ? 'pointer' : '';
+
+      if (!features.length) {
+        popup.remove();
+        return;
       }
     })
+
     map.on('click', function (e) {
-      var features = map.queryRenderedFeatures(
-        [e.point.x, e.point.y],
-        {layers: ['tm-projects-black-circle', 'tm-projects-symbol']}
-      )
-      if (features.length) {
-        $('#proj-details').empty()
-        $('#proj-details').append(
-          '<p style="font-weight:bold" id = "details-mappers">Mappers: ' + formatedData(features[0].properties.mappers) + '</p>' +
-          '<p style="font-weight:bold" id= "details-edits">Edits: ' + formatedData(features[0].properties.edits) + '</p>'
-        )
+      var features = map.queryRenderedFeatures(e.point, {
+        layers: ['tm-projects-black-circle', 'tm-projects-symbol']
+      });
+
+      if (!features.length) {
+        return;
       }
-    })
+
+      var feature = features[0];
+
+      updatePopupOnClick(feature);
+    });
   }
+
   var projectExtentJSON
   const downloadUrl = proxyUrl + driveUrl + fileId
   $.get(downloadUrl, function (data) {
     projectExtentJSON = JSON.parse(data)
-    var bbox = turf.bbox(projectExtentJSON.features[0])
-    // set bounds according to features
-    map.fitBounds(bbox, {
-      padding: 50,
-      maxZoom: 14.15,
-      duration: 2000
-    })
+
+    projectExtentJSON.features.forEach(function (feature) {
+      bboxCoordinatesArray.push(turf.bbox(feature));
+    });
+
+    var numOfPolygons = projectExtentJSON.features.length;
+
+    if (numOfPolygons.length === 1) {
```

Comment:
Quick point- `numOfPolygons` already stores the length of the array. So the condition is `numOfPolygons.length` is invalid. Also you could use length only on array variables. Just using `numOfPolygons === 1` should work

Reply:
I ran through the code with :
```
if (numOfPolygons === 1) {
      setMapView();
}
```
This returns a console error: `camera.js:415 Uncaught TypeError: Cannot read property 'getNorthWest' of undefined`. This is because `currentPolygonIndex` is initialised to zero after this block. This will work when you declare the variable with an intial value of `0`

Reply:
Fly button should be displayed only if `numOfPolygons` > 1

Reply:
My mistake, sorry.

Reply:
Great - button displays for just projects with more than one polygons